charmcraft/commands/init.py

```diff
@@ -92,7 +92,11 @@ def run(self, args):
         logger.debug("Using project directory '%s'", self.config.project.dirpath)
 
         if args.author is None:
-            gecos = pwd.getpwuid(os.getuid()).pw_gecos.split(",", 1)[0]
+            try:
+                gecos = pwd.getpwuid(os.getuid()).pw_gecos.split(",", 1)[0]
+            except KeyError:
+                # no info for the user
+                gecos = None
```

Comment:
if gecos is only ever assigned in this try block, why not directly raise the CommandError here?